opendebates/admin.py

```diff
@@ -103,3 +104,8 @@ class SiteModeAdmin(ModelAdmin):
     _missed_fields = set(_fields) - _used_fields
     if _missed_fields:
         fieldsets.append(('Other', {'fields': sorted(list(_missed_fields))}))
+
+    def reset_current_votes(self, request, queryset):
+        # Note, currently all submissions are under the one and only SiteMode.
+        models.Submission.objects.update(current_votes=0)
```

Comment:
@ejucovy yes, you are correct, 1 & 2 would have to be executed in the correct order.  I've updated the other locations to do your suggested check for the `previous_debate_time` still being in the future.